compiler/mir/mirtrees.nim

```diff
@@ -56,166 +56,138 @@ type
+                ## destination is empty
+
+    mnkSwitch ## sets the value of a discriminator field, changing the active
+              ## branch, if necessary. The destination operand must be a
+              ## ``mnkPathVariant`` expression
 
     mnkPathNamed ## access of a named field in a record
     mnkPathPos   ## access of a field in record via its position
-    mnkPathArray ## ``pathArray(x, i)``; array-like access
-    mnkPathVariant ## access a field inside a tagged union
+    # future direction: merge ``mnkPathPos`` with ``mnkPathNamed``. This first
+    # requires a dedicated MIR type representation.
+    mnkPathArray ## access of an array-like (both dynamic and static) value
+                 ## with an integer index
+    mnkPathVariant ## access a tagged union
                    ## XXX: this is likely only a temporary solution. Each
                    ##      record-case part of an object should be its own
                    ##      dedicated object type, which can then be addressed
                    ##      as a normal field
-    mnkPathConv  ## an handle conversion. That is, a conversion that produces a
+    # future direction: merge ``mnkPathVariant`` into ``mnkPathNamed`` once the
+    # MIR's record type structure supports this
+    mnkPathConv  ## a handle conversion. That is, a conversion that produces a
                  ## *handle*, and not a new *value*. At present, this operator
                  ## also applies to first-class handles, like ``ref``.
 
-    mnkAddr   ## ``addr(x)``; creates a first-class unsafe alias/handle (i.e.
-              ## pointer) from the input lvalue `x`
-    mnkDeref  ## ``deref(x)``; dereferences the pointer-like `x` (this
-              ## *excludes* views), producing an lvalue that has same identity
-              ## as the pointed-to location
-              ## XXX: the possibility of invalid pointers (e.g. nil pointer) is
-              ##      currently ignored
-
-    # XXX: the exact semantics around views and their related operators are
-    #      not yet finalized. One should not rely on them too much at this
-    #      point
-    mnkView      ## ``view(x)``; creates a safe alias of the l-value 'x'
-    mnkDerefView ## ``derefView(x)``; dereferences a view, producing the lvalue
-                 ## named by it
-    # XXX: ``mnkDerefView`` is not used for ``openArray`` right now, due to
-    #      the latter's interactions with ``var`` and ``lent``
-
-    mnkStdConv    ## ``stdConv(x)``; a standard conversion.Produce a new value.
-                  ## Also used for to-slice (``openArray``) conversions, in
-                  ## which case the semantics are still fuzzy.
+    mnkAddr   ## create a pointer from the provided lvalue
+    mnkDeref  ## dereference a ``ptr`` or ``ref`` value
+
+    mnkView      ## create a first-class safe alias from an lvalue
+    mnkDerefView ## dereference a first-class safe alias
+
+    mnkStdConv    ## a standard conversion. Produce a new value.
```

Comment:
Is standard here just meant built-in?

Reply:
Essentially, yes. It's terminology I reused from `PNode` (i.e., `nkHiddenStdConv` vs. `nkConv`).

In any case, the distinction is not needed anymore, and I'm going to remove `mnkStdConv` in a follow-up.